AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_DirectionalLightAdded.py

```diff
@@ -146,37 +325,452 @@ def AtomEditorComponents_DirectionalLight_AddedToEntity():
+                    AtomComponentProperties.directional_light('Angular diameter')), 0.5))
+
+        # 12. Add Camera entity.
         camera_entity = EditorEntity.create_editor_entity(AtomComponentProperties.camera())
         Report.result(Tests.camera_creation, camera_entity.exists())
 
-        # 9. Add Camera component to Camera entity.
+        # 13. Add Camera component to Camera entity.
         camera_entity.add_component(AtomComponentProperties.camera())
         Report.result(Tests.camera_component_added, camera_entity.has_component(AtomComponentProperties.camera()))
 
-        # 10. Set the Directional Light component property Shadow|Camera to the Camera entity.
+        # 14. Set the Directional Light component property Shadow|Camera to the Camera entity.
         directional_light_component.set_component_property_value(
             AtomComponentProperties.directional_light('Camera'), camera_entity.id)
         Report.result(
             Tests.shadow_camera_check,
             camera_entity.id == directional_light_component.get_component_property_value(
                                     AtomComponentProperties.directional_light('Camera')))
 
-        # 11. Delete DirectionalLight entity.
+        # 15. Shadow Far Clip set to 2000 then back to the default of 100.
+        directional_light_component.set_component_property_value(
+            AtomComponentProperties.directional_light('Shadow far clip'), 2000.0)
+        Report.result(
+            Tests.intensity_set_to_10,
+            directional_light_component.get_component_property_value(
+                AtomComponentProperties.directional_light('Shadow far clip')) == 2000.0)
+
+        directional_light_component.set_component_property_value(
+            AtomComponentProperties.directional_light('Shadow far clip'), 100.0)
+        general.idle_wait_frames(1)
+        Report.result(
+            Tests.intensity_set_to_default,
+            directional_light_component.get_component_property_value(
+                AtomComponentProperties.directional_light('Shadow far clip')) == 100.0)
+
+        # 16. Shadow Map size set to 256 then back to the default of 1024.
+        directional_light_component.set_component_property_value(
+            AtomComponentProperties.directional_light('Shadowmap size'), 256)
+        Report.result(
+            Tests.shadowmap_size_set_to_256,
+            directional_light_component.get_component_property_value(
+                AtomComponentProperties.directional_light('Shadowmap size')) == 256)
+
+        directional_light_component.set_component_property_value(
+            AtomComponentProperties.directional_light('Shadowmap size'), 1024)
+        general.idle_wait_frames(1)
+        Report.result(
+            Tests.shadowmap_size_set_to_default,
+            directional_light_component.get_component_property_value(
+                AtomComponentProperties.directional_light('Shadowmap size')) == 1024)
```

Comment:
Shadowmap size is an enum with specific integer mapping so this works, but other integer values might not. you don't strictly need to change this, but maybe the docstring or comments here can call out that only a small set of values are valid.

        AZ_ENUM_CLASS_WITH_UNDERLYING_TYPE(ShadowmapSize, uint32_t,
            // For None, shadowmap image has width=1 and height=1.
            // Several passes require the image resource so, we minimize its memory footprint.
            (None, 1), // disabling shadowmap
            (Size256, 256),
            (Size512, 512),
            (Size1024, 1024),
            (Size2048, 2048));

Reply:
Fixed

Reply:
1 is also a valid value which is None disabling shadowmap.